lib/system/iterators_1.nim

```diff
@@ -23,22 +23,24 @@ iterator countdown*[T](a, b: T, step: Positive = 1): T {.inline.} =
       for i in countdown(9, 2, 3):
         i
     assert y == @[9, 6, 3]
-  when T is (uint|uint64):
-    var res = a
-    while res >= b:
-      yield res
-      if res == b: break
-      dec(res, step)
-  elif T is IntLikeForCount and T is Ordinal:
-    var res = int(a)
-    while res >= int(b):
-      yield T(res)
-      dec(res, step)
-  else:
-    var res = a
-    while res >= b:
-      yield res
-      dec(res, step)
+  try:
+    when T is (uint|uint64):
+      var res = a
+      while res >= b:
+        yield res
+        if res == b: break
+        dec(res, step)
+    elif T is IntLikeForCount and T is Ordinal:
+      var res = int(a)
+      while res >= int(b):
+        yield T(res)
+        dec(res, step)
+    else:
+      var res = a
+      while res >= b:
+        yield res
+        dec(res, step)
+  except RangeDefect, OverflowDefect: discard
```

Comment:
Yeah, as @Varriount said, code like that won't be approved into the stdlib, because Defects are supposed to be program errors that are bugs, and you can't catch them if you compile with `--panics:on`.